commit 653c10b4097fc043308856d2092f8d6569fd87ba
Author: Tim Kosse <tim.kosse@filezilla-project.org>
Date:   Mon Sep 4 14:56:03 2017 +0200

    Need to unbind events if destroying wxTextEntryHintData, otherwise there will be segfaults.

diff --git a/src/common/textentrycmn.cpp b/src/common/textentrycmn.cpp
index b1480e322e..1d70ecd6f6 100644
--- a/src/common/textentrycmn.cpp
+++ b/src/common/textentrycmn.cpp
@@ -54,7 +54,14 @@ public:
                                 wxTextEntryHintData::OnTextChanged, this);
     }
 
-    // default dtor is ok
+#ifdef wxHAS_EVENT_BIND
+	~wxTextEntryHintData()
+	{
+		m_win->Unbind(wxEVT_SET_FOCUS, &wxTextEntryHintData::OnSetFocus, this);
+		m_win->Unbind(wxEVT_KILL_FOCUS, &wxTextEntryHintData::OnKillFocus, this);
+		m_win->Unbind(wxEVT_TEXT, &wxTextEntryHintData::OnTextChanged, this);
+	}
+#endif
 
     // Get the real text of the control such as it was before we replaced it
     // with the hint.
